CI: minor fixes to tox config

bsipocz opened the pull request:
Minor changes, I think only the first one is controversial, I opted into building the dev version, but it's probably not worth the added extra minutes:

- scikit-image doesn't seem to have nightly wheels. We can either build a dev version or only test releases.
- mpl has nightly
- pip freeze once we installed everything
- duplicate conftest, so CI will show the custom test header

bsipocz commented:
I'm not exactly sure what goes wrong with the CI job, I'll try to investigate later.

larrybradley commented:
I'd like to get the tox updates merged.  Can you modify to PR with only the tox updates and submit a followup PR for the `conftest.py`.  IIRC, I've tried moving `conftest` before, but it never worked.

bsipocz commented:
I figured out a lot of things about conftest in the past two weeks, so will come back to this and fix it up in the evening.

bsipocz commented:
OK, so I removed the conftest. I don't see why it prevents the package build for tox, and I couldn't spot the difference compared to other packages where it works.
With using pip freeze, getting the header right is actually not as helpful as it once was, so I removed it from this PR.

larrybradley reviewed (approved):
Thanks, @bsipocz!
